runtime/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSessionFactory.java

```diff
@@ -3292,10 +3323,24 @@ private void onKafkaReset(
             final long sequence = reset.sequence();
             final long acknowledge = reset.acknowledge();
             final long traceId = reset.traceId();
+            final OctetsFW extension = reset.extension();
 
             assert acknowledge <= sequence;
 
-            delegate.doMqttReset(traceId, EMPTY_OCTETS);
+
+            final KafkaResetExFW kafkaResetEx = extension.get(kafkaResetExRO::tryWrap);
+            final int error = kafkaResetEx != null ? kafkaResetEx.error() : -1;
+
+            Flyweight mqttResetEx = EMPTY_OCTETS;
+            if (error != -1)
+            {
+                mqttResetEx = mqttSessionResetExRW.wrap(sessionExtBuffer, 0, sessionExtBuffer.capacity())
+                    .typeId(mqttTypeId)
+                    .reasonCode(MQTT_REASON_CODES.get(kafkaResetEx.error()) & 0xff)
+                    .reason(MQTT_REASONS.getOrDefault(kafkaResetEx.error(), EMPTY_STRING))
```

Comment:
Why do we default to `EMPTY_STRING` here?
Should be `null`, and then `reason` string should be omitted from the encoded mqtt packet when `null`, right?